src/domain/exceptions.py

```
@@ -11,6 +11,14 @@ class DataSourceException(AirQualityException):
     pass
 
 
+class DataSourceError(DataSourceException):
+    pass
+
+
+class APIError(DataSourceException):
```

Pylint (3.3.7)

[convention] Missing class docstring

(C0115)